std/valgrind/index.zig

```diff
@@ -0,0 +1,367 @@
+const builtin = @import("builtin");
+const math = @import("index.zig").math;
+
+
+const noValgrind = builtin.mode == builtin.Mode.ReleaseFast or
+        builtin.mode == builtin.Mode.ReleaseSmall;
+
+
+pub fn doClientRequest(default: usize, request: usize,
+        a1: usize, a2: usize, a3: usize, a4: usize, a5: usize
+    ) usize
+{
+    if (noValgrind) {
+        return default;
+    }
+
+    switch (builtin.arch) {
+        builtin.Arch.i386 => {
+            return asm volatile (
+                \\ roll $3,  %%edi ; roll $13, %%edi
+                \\ roll $29, %%edi ; roll $19, %%edi
+                \\ xchgl %%ebx,%%ebx
+                : [_] "={edx}" (-> usize)
+                : [_] "{eax}" ([]usize{request, a1, a2, a3, a4, a5}),
```

Comment:
@andrewrk I'm surprised this didn't need `.ptr` or some sort of cast to go from array to pointer. Is there something else I should be using here?

Reply:
FWIW inline asm arguments are not really checked by the compiler yet, you should probably inspect the generated asm yourself to be sure.